### DISCUSSION TITLE
Extending Input Map Menu with a refresh button

### DISCUSSION THREAD

**Discussion by Err-godic** [godotengine/godot-proposals#7384] [Editor]
I am working on a plugin that modifies key bindings in ProjectSettings.  Here is a snippet of the relevant GDScript to give you an idea.

```
var event_list = InputMap.action_get_events(KeyzMenu.pressed_event_action)
InputMap.action_erase_events(KeyzMenu.pressed_event_action)
		
var setting_string: String = "input/" + KeyzMenu.pressed_event_action
var dict = ProjectSettings.get_setting(setting_string)
	
for e in event_list:
	if (e.as_text() == KeyzMenu.pressed_event):
			e = event
	InputMap.action_add_event(KeyzMenu.pressed_event_action, e)
			
event_list = InputMap.action_get_events(KeyzMenu.pressed_event_action)
dict["events"] = event_list
			
ProjectSettings.set_setting(setting_string, dict)
```

So in the plugin note how any changes I make don't update until the third image. In the third image, I toggle "Show Built-In Actions" to force the menu to refresh changes.

![07-26-2023-104826](https://github.com/godotengine/godot-proposals/assets/114266588/715ab24a-81db-4b3a-aeec-a452bb26636b)

![07-26-2023-104903](https://github.com/godotengine/godot-proposals/assets/114266588/fe97c8f1-3b2b-4d3f-8f47-7f9c250103cc)

![07-26-2023-104916](https://github.com/godotengine/godot-proposals/assets/114266588/076393a2-3283-4b52-bd17-3ac0efbea731)

Under the action "Bacon", "Mouse Wheel Up" gets replaced with "Middle Mouse Button" after I do this toggle.

Maybe asking for a way to refresh all ProjectSettings from a script with a single function may be asking too much, but I think having some kind of refresh button in the editor or functions for specific settings would be reasonable (in particular, the input maps). 

What do you all think?

**Comment by Calinou** [comment]
- This is actually a bug: https://github.com/godotengine/godot/issues/25865

We should fix the bug, so that we don't need to add a Refresh button.

**Reply by Err-godic** [reply]
Ah I see, thanks for the reply. Not sure how much it matters but I should mention I'm using 4.1.1 x86_64 Linux.
